packages/react-components/react-aria/src/AriaLiveAnnouncer/useDomAnnounce.ts

```diff
@@ -0,0 +1,151 @@
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import { createPriorityQueue, useTimeout } from '@fluentui/react-utilities';
+import * as React from 'react';
+
+import type { AriaLiveAnnounceFn, AriaLiveMessage } from './AriaLiveAnnouncer.types';
+
+/** The duration the message needs to be in present in DOM for screen readers to register a change and announce */
+const MESSAGE_DURATION = 500;
+
+const VISUALLY_HIDDEN_STYLES = {
+  clip: 'rect(0px, 0px, 0px, 0px)',
+  height: '1px',
+  margin: '-1px',
+  width: '1px',
+  position: 'absolute',
+  overflow: 'hidden',
+  textWrap: 'nowrap',
+};
+
+/* INTERNAL: implementation of the announcer using a live region element */
+export const useDomAnnounce_unstable = (): AriaLiveAnnounceFn => {
+  const { targetDocument } = useFluent();
+
+  const timeoutRef = React.useRef<number | undefined>(undefined);
+  const [setAnnounceTimeout, clearAnnounceTimeout] = useTimeout();
+
+  const elementRef = React.useRef<HTMLDivElement | null>(null);
+
+  const order = React.useRef(0);
+
+  // investigate alert implementation later
+  // const [alertList, setAlertList] = React.useState<string[]>([]);
+
+  const batchMessages = React.useRef<{ batchId: string; message: AriaLiveMessage }[]>([]);
+
+  const [messageQueue] = React.useState(() =>
+    createPriorityQueue<AriaLiveMessage>((a, b) => {
+      if (a.priority !== b.priority) {
+        return b.priority - a.priority;
+      }
+
+      return a.createdAt - b.createdAt;
+    }),
+  );
+
+  const queueMessage = React.useCallback(() => {
+    if (timeoutRef.current || !elementRef.current) {
+      return;
+    }
+
+    const runCycle = () => {
+      if (!elementRef.current) {
+        return;
+      }
+
+      if (targetDocument && messageQueue.peek()) {
+        // need a wrapping element for Narrator/Edge, which currently does not pick up text-only live region changes
+        // consistently
+        // if this is fixed, we can set textContent to the string directly
+
+        const wrappingEl = targetDocument.createElement('span');
+
+        wrappingEl.innerText = messageQueue
+          .all()
+          .filter(msg => msg.message.trim().length > 0)
+          .reduce((prevText, currMsg) => prevText + currMsg.message + '. ', '');
+
+        elementRef.current.innerText = '';
+        elementRef.current.appendChild(wrappingEl);
+
+        messageQueue.clear();
+        batchMessages.current = [];
+
+        // begin new cycle to clear (or update) messages
+        timeoutRef.current = setAnnounceTimeout(() => {
+          runCycle();
+        }, MESSAGE_DURATION);
+      } else {
+        elementRef.current.textContent = '';
+        clearAnnounceTimeout();
```

Comment:
I think you'll want an effect that calls this and nulls out `timeoutRef` in the case that this hook is unmounted with a pending timer.

Reply:
Ah good point -- I think it could be added to the `useEffect` that currently sets up `elementRef` and has a teardown fn.

I think I'd lean towards doing this in a separate PR if you're also OK with that -- right now `useDomAnnounce` is a straight copy-paste of what's currently in `useAriaLiveAnnouncer`, and I'm hesitant to introduce additional changes to this PR, since it's already a reasonable chunk to review.

Reply:
Sounds good to me.